lib/Sema/SemaExpr.cpp

```diff
@@ -13465,170 +13465,59 @@ bool Sema::CheckBoundsCastBaseType(Expr *E1) {
 ExprResult Sema::ActOnBoundsCastExprBounds(
     Scope *S, SourceLocation OpLoc, tok::TokenKind Kind,
     SourceLocation LAngleBracketLoc, ParsedType D,
-    SourceLocation RAngleBracketLoc, RelativeBoundsClause *RelativeClause,
+    SourceLocation RAngleBracketLoc,
     SourceLocation LParenLoc, SourceLocation RParenLoc, Expr *E1,
-    Expr *bounds) {
+    BoundsExpr *Bounds) {
+  TypeSourceInfo *CastTInfo;
 
-  RangeBoundsExpr *Range = nullptr;
-  TypeSourceInfo *castTInfo;
-
-  QualType DestTy = GetTypeFromParser(D, &castTInfo);
-  ExprResult ParsedBounds = CorrectDelayedTyposInExpr(bounds);
-  SourceLocation TypeLoc = (castTInfo->getTypeLoc()).getBeginLoc();
+  QualType DestTy = GetTypeFromParser(D, &CastTInfo);
```

Comment:
I looked at the code for `GetTypeFromParser` and I think that either way, it will set `CastTInfo` to `nullptr`, so it's okay that it's declared un-initialized. Maybe there should be an `assert` before it's de-referenced, though?